src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,512 @@
+    EmptyEnumOption,
+
+    /// Empty optional field.
+    #[error("the optional field `{0}` is empty and should be removed")]
+    EmptyOptionalField(String),
+}
+
+/// Arguments that support a fixed set of declared values as constant arguments.
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumArgument {
+    /// A human-readable name for this argument to help clarify use.
+    name: String,
+
+    /// Additional description for this argument.
+    description: String,
+
+    /// List of valid string options for this argument.
+    options: Vec<String>,
+}
+
+impl EnumArgument {
+    fn parse(
+        name: Option<String>,
+        description: Option<String>,
+        options: Vec<String>,
+    ) -> Result<EnumArgument, ArgumentsItemError> {
+        if options.iter().any(String::is_empty) {
+            return Err(ArgumentsItemError::EmptyEnumOption);
+        }
+
+        Ok(EnumArgument {
+            name: ArgumentsItem::parse_name(name)?,
+            description: ArgumentsItem::parse_description(description)?,
+            options,
+        })
+    }
+}
+
+impl From<EnumArgument> for ArgumentsItem {
+    fn from(value: EnumArgument) -> Self {
+        ArgumentsItem::EnumArgument(value)
+    }
+}
+
+/// Arguments that refer to a data value.
+#[derive(Clone, Debug)]
+pub struct ValueArgument {
+    /// A human-readable name for this argument to help clarify use.
+    name: String,
+
+    /// Additional description for this argument.
+    description: String,
+
+    /// A fully defined type or a type expression.
+    ///
+    /// todo: implement parsed [simple_extensions::Type].
+    value: simple_extensions::Type,
+
+    /// Whether this argument is required to be a constant for invocation.
+    /// For example, in some system a regular expression pattern would only be accepted as a literal
+    /// and not a column value reference.
+    constant: bool,
+}
+
+impl ValueArgument {
+    fn parse<C: Context>(
+        // Retain this parameter for future value parsing
+        _ctx: &mut C,
+        name: Option<String>,
+        description: Option<String>,
+        value: simple_extensions::Type,
+        constant: Option<bool>,
+    ) -> Result<ValueArgument, ArgumentsItemError> {
+        Ok(ValueArgument {
+            name: ArgumentsItem::parse_name(name)?,
+            description: ArgumentsItem::parse_description(description)?,
+            value: value,
+            constant: constant.unwrap_or_default(),
+        })
+    }
+}
+
+impl From<ValueArgument> for ArgumentsItem {
+    fn from(value: ValueArgument) -> Self {
+        ArgumentsItem::ValueArgument(value)
+    }
+}
+
+/// Arguments that are used only to inform the evaluation and/or type derivation of the function.
+#[derive(Clone, Debug, PartialEq)]
+pub struct TypeArgument {
+    /// A human-readable name for this argument to help clarify use.
+    name: String,
+
+    /// Additional description for this argument.
+    description: String,
```

Comment:
These fields aren't required:
```suggestion
    /// A human-readable name for this argument to help clarify use.
    name: Option<String>,

    /// Additional description for this argument.
    description: Option<String>,
```

Reply:
same as above.